pkg/controller/network/networkinterface/managed.go

```diff
@@ -0,0 +1,147 @@
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha3.NetworkInterface{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha3.NetworkInterfaceGroupVersionKind),
+			managed.WithConnectionPublishers(),
+			managed.WithExternalConnecter(&connecter{client: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connecter struct {
+	client client.Client
+}
+
+func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	creds, auth, err := azureclients.GetAuthInfo(ctx, c.client, mg)
+	if err != nil {
+		return nil, err
+	}
+	cl := azurenetwork.NewInterfacesClient(creds[azureclients.CredentialsKeySubscriptionID])
+	cl.Authorizer = auth
+	return &external{client: cl}, nil
+}
+
+type external struct {
+	client networkapi.InterfacesClientAPI
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	s, ok := mg.(*v1alpha3.NetworkInterface)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotNetworkInterface)
+	}
+
+	az, err := e.client.Get(ctx, s.Spec.ResourceGroupName, meta.GetExternalName(s), "")
+	if azureclients.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetNetworkInterface)
+	}
+
+	network.UpdateNetworkInterfaceStatusFromAzure(s, az)
+	s.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	s, ok := mg.(*v1alpha3.NetworkInterface)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotNetworkInterface)
+	}
+
+	s.Status.SetConditions(xpv1.Creating())
+
+	snet := network.NewNetworkInterfaceParameters(s)
+	if _, err := e.client.CreateOrUpdate(ctx, s.Spec.ResourceGroupName, meta.GetExternalName(s), snet); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateNetworkInterface)
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	_, ok := mg.(*v1alpha3.NetworkInterface)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotNetworkInterface)
+	}
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+	s, ok := mg.(*v1alpha3.NetworkInterface)
+	if !ok {
+		return errors.New(errNotNetworkInterface)
+	}
+
+	mg.SetConditions(xpv1.Deleting())
```

Comment:
crossplane-runtime can now [handle setting "creating" and "deleting" conditions](https://github.com/crossplane/crossplane-runtime/pull/242).